ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/signing/hashicorp/HashicorpSignerHelper.java

```diff
@@ -0,0 +1,138 @@
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.JsonObject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.web3j.crypto.Credentials;
+
+public class HashicorpSignerHelper {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final Vertx vertx = Vertx.vertx();
+  private static final HttpClient hashicorpVaultClient = vertx.createHttpClient();
+
+  private HashicorpSignerHelper() {}
+
+  public static CredentialTransactionSigner getSigner(HashicorpSignerConfig config) {
+    final String tokenString = readTokenFromFile(config);
+    if (tokenString == null) {
+      return null;
+    }
+    final String response = requestSecretFromVault(config, tokenString);
+    if (response == null) {
+      return null;
+    }
+    final Credentials credentials = extractCredentialsFromJson(response);
+    if (credentials == null) {
+      return null;
+    }
+    return new CredentialTransactionSigner(credentials);
+  }
+
+  private static String readTokenFromFile(HashicorpSignerConfig config) {
+    List<String> authFileLines;
+    try {
+      authFileLines = Files.readAllLines(config.getAuthFilePath());
+    } catch (IOException e) {
+      LOG.error(
+          "Unable to read file containing the authentication information for Hashicorp Vault.\n"
+              + config.getAuthFilePath()
+              + "\n"
+              + e.getMessage());
+      return null;
+    }
+    return authFileLines.get(0);
+  }
+
+  private static String requestSecretFromVault(HashicorpSignerConfig config, String tokenString) {
+    final String requestURI = "/v1" + config.getSigningKeyPath();
+
+    return getVaultResponse(config, tokenString, requestURI);
+  }
+
+  private static String getVaultResponse(
+      HashicorpSignerConfig config, String tokenString, String requestURI) {
+    final CompletableFuture<String> future = new CompletableFuture<>();
+    final HttpClientRequest request =
+        hashicorpVaultClient.request(
+            HttpMethod.GET,
+            config.getServerPort(),
+            config.getServerHost(),
+            requestURI,
+            rh -> {
+              rh.bodyHandler(
+                  bh -> {
+                    if (rh.statusCode() == 200) {
+                      future.complete(bh.toString());
+                    } else {
+                      future.completeExceptionally(
+                          new Exception(
+                              "Hashicorp vault responded with status code {"
+                                  + rh.statusCode()
+                                  + "}"));
+                    }
+                  });
+            });
+    if (tokenString != null) {
+      request.headers().set("X-Vault-Token", tokenString);
+    }
+    request.setChunked(false);
+    request.end();
+    String response;
+    try {
+      response = future.get(config.getTimeout(), TimeUnit.SECONDS);
```

Comment:
maybe better putting the timeout on the Request, then having an Exception handler attached to the request (see PassThroughHandler for an example).